store/store.go

```diff
@@ -1178,6 +1199,93 @@ func (s *Store) Find(ctx context.Context, search *Search, user *auth.UserState)
 		return nil, ErrBadQuery
 	}
 
+	q := url.Values{}
+	q.Set("fields", strings.Join(s.searchFields, ","))
+	q.Set("architecture", s.architecture)
+
+	if search.Private {
+		q.Set("private", "true")
+	}
+
+	if search.Prefix {
+		q.Set("name", searchTerm)
+	} else {
+		if search.CommonID != "" {
+			q.Set("common-id", search.CommonID)
+		}
+		if searchTerm != "" {
+			q.Set("q", searchTerm)
+		}
+	}
+
+	// section is "category" in search v2
+	if search.Section != "" {
+		q.Set("category", search.Section)
+	}
+
+	// with search v2 all risks are searched by default (same as scope=wide
+	// with v1) so we need to restrict channel if scope is not passed.
+	if search.Scope == "" {
```

Comment:
This would need to become:
```
if search.Scope != "wide" {
  q.Set("channel", "stable")
}
```
wide is the default in v2. cmd_find passes empty scope if `--narrow` is requested.

Is that what you had in mind? Does it make sense?

Reply:
yes, but I think we should also return BadRequest if Scope is not "wide" or ""

Reply:
Done (ErrInvalidScope returned).